state/cloudcredentials_test.go

```diff
@@ -112,6 +114,118 @@ func (s *CloudCredentialsSuite) TestUpdateCloudCredentialsExisting(c *gc.C) {
 	c.Assert(out, jc.DeepEquals, expected)
 }
 
+func assertCredentialCreated(c *gc.C, testSuite ConnSuite) (string, *state.User, names.CloudCredentialTag) {
+	owner := testSuite.Factory.MakeUser(c, &factory.UserParams{
+		Password: "secret",
+		Name:     "bob",
+	})
+
+	cloudName := "stratus"
+	err := testSuite.State.AddCloud(cloud.Cloud{
+		Name:      cloudName,
+		Type:      "low",
+		AuthTypes: cloud.AuthTypes{cloud.AccessKeyAuthType, cloud.UserPassAuthType},
+		Regions:   []cloud.Region{{Name: "dummy-region", Endpoint: "endpoint"}},
+	}, owner.Name())
+	c.Assert(err, jc.ErrorIsNil)
+
+	tag := createCredential(c, testSuite, cloudName, owner.Name(), "foobar")
+	return cloudName, owner, tag
+}
+
+func createCredential(c *gc.C, testSuite ConnSuite, cloudName, userName, credentialName string) names.CloudCredentialTag {
+	cred := cloud.NewCredential(cloud.AccessKeyAuthType, map[string]string{
+		"foo": "foo val",
+		"bar": "bar val",
+	})
+	tag := names.NewCloudCredentialTag(fmt.Sprintf("%v/%v/%v", cloudName, userName, credentialName))
+	err := testSuite.State.UpdateCloudCredential(tag, cred)
+	c.Assert(err, jc.ErrorIsNil)
+	return tag
+}
+
+func assertModelCreated(c *gc.C, testSuite ConnSuite, cloudName string, credentialTag names.CloudCredentialTag, owner names.Tag, modelName string) string {
+	// Test model needs to be on the test cloud for all validation to pass.
+	modelState := testSuite.Factory.MakeModel(c, &factory.ModelParams{
+		Name:            modelName,
+		CloudCredential: credentialTag,
+		Owner:           owner,
+		CloudName:       cloudName,
+	})
+	defer modelState.Close()
+	testModel, err := modelState.Model()
+	c.Assert(err, jc.ErrorIsNil)
+	assertModelStatus(c, testSuite.StatePool, testModel.UUID(), status.Available)
+	return testModel.UUID()
+}
+
+func assertModelSuspended(c *gc.C, testSuite ConnSuite) (names.CloudCredentialTag, string) {
+	// 1. Create a credential
+	cloudName, credentialOwner, credentialTag := assertCredentialCreated(c, testSuite)
+
+	// 2. Create model on the test cloud with test credential
+	modelUUID := assertModelCreated(c, testSuite, cloudName, credentialTag, credentialOwner.Tag(), "model-for-cloud")
+
+	// 3. update credential to be invalid and check model is suspended
+	cred := cloud.NewCredential(cloud.AccessKeyAuthType, map[string]string{
+		"foo": "foo val",
+		"bar": "bar val",
+	})
+	cred.Invalid = true
+	cred.InvalidReason = "because it is really really invalid"
+	err := testSuite.State.UpdateCloudCredential(credentialTag, cred)
+	c.Assert(err, jc.ErrorIsNil)
+	assertModelStatus(c, testSuite.StatePool, modelUUID, status.Suspended)
```

Comment:
can we also check (here or elsewhere) the suspended status info string as well?

Reply:
Yes